pkg/utils/utils.go

```diff
@@ -0,0 +1,20 @@
+package utils
+
+import (
+        "crypto/sha512"
+        "fmt"
+)
+
+const maxChainName = 29
+
+// Generates a chain name to be used with iptables.
+// Ensures that the generated name is less than
+// maxChainName chars in length
+func FormatChainName(name string, id string) string {
+        h := sha512.Sum512([]byte(id))
+        chain := fmt.Sprintf("CNI-%s-%x", name, h[:8])
```

Comment:
What about using `name[:16]` to ensure that `h` will not be completely trimmed if `name` is very long?